client/lib/analytics/index.js

```diff
@@ -386,117 +290,15 @@ const analytics = {
 
 	tracks: {
 		recordEvent: function( eventName, eventProperties ) {
-			eventProperties = eventProperties || {};
-
-			if ( process.env.NODE_ENV !== 'production' && typeof console !== 'undefined' ) {
-				if (
-					! /^calypso(?:_[a-z]+){2,}$/.test( eventName ) &&
-					! includes( EVENT_NAME_EXCEPTIONS, eventName )
-				) {
-					//eslint-disable-next-line no-console
-					console.error(
-						'Tracks: Event `%s` will be ignored because it does not match /^calypso(?:_[a-z]+){2,}$/ and is ' +
-							'not a listed exception. Please use a compliant event name.',
-						eventName
-					);
-				}
-
-				for ( const key in eventProperties ) {
-					if ( isObjectLike( eventProperties[ key ] ) ) {
-						const errorMessage =
-							`Tracks: Unable to record event "${ eventName }" because nested ` +
-							`properties are not supported by Tracks. Check '${ key }' on`;
-						console.error( errorMessage, eventProperties ); //eslint-disable-line no-console
-						return;
-					}
-
-					if ( ! /^[a-z_][a-z0-9_]*$/.test( key ) ) {
-						//eslint-disable-next-line no-console
-						console.error(
-							'Tracks: Event `%s` will be rejected because property name `%s` does not match /^[a-z_][a-z0-9_]*$/. ' +
-								'Please use a compliant property name.',
-							eventName,
-							key
-						);
-					}
-
-					if ( TRACKS_SPECIAL_PROPS_NAMES.indexOf( key ) !== -1 ) {
-						//eslint-disable-next-line no-console
-						console.error(
-							"Tracks: Event property `%s` will be overwritten because it uses one of Tracks' internal prop name: %s. " +
-								'Please use another property name.',
-							key,
-							TRACKS_SPECIAL_PROPS_NAMES.join( ', ' )
-						);
-					}
-				}
-			}
-
-			tracksDebug( 'Record event "%s" called with props %o', eventName, eventProperties );
-
-			if (
-				! eventName.startsWith( 'calypso_' ) &&
-				! includes( EVENT_NAME_EXCEPTIONS, eventName )
-			) {
-				tracksDebug(
-					'- Event name must be prefixed by "calypso_" or added to `EVENT_NAME_EXCEPTIONS`'
-				);
-				return;
-			}
-
-			if ( _superProps ) {
-				const superProperties = _superProps( eventProperties );
-				eventProperties = { ...eventProperties, ...superProperties }; // assign to a new object so we don't modify the argument
-			}
-
-			// Remove properties that have an undefined value
-			// This allows a caller to easily remove properties from the recorded set by setting them to undefined
-			eventProperties = omit( eventProperties, isUndefined );
-
-			tracksDebug( 'Recording event "%s" with actual props %o', eventName, eventProperties );
+			analyticsEvents.once( 'record-event', ( _eventName, _eventProperties ) => {
+				analytics.emit( 'record-event', _eventName, _eventProperties );
+			} );
```

Comment:
The `record-event` event has only one usage: reporting the last tracks event as part of debug info when the JS app crashes.

That could be replaced with a `getLastTracksEvent()` function that's part of the tracks API.

And maybe it could be removed altogether -- I'm not sure if we use that info for anything and I don't see why it would be interesting when debugging JS crashes.

Reply:
Hmm I'd rather not take this out in this PR, so the PR can stay pure refactoring.

Reply:
I did investigate it and I believe adding the `getLastTracksEvent` would work but I saw it was possible that other events are added also using the `recordEvent` name so the behavior might change slightly.. It probably wouldn't be an issue to anyone but I don't want to take on the burden of testign it etc.